### PR TITLE
Variant picker improvements

### PR DESCRIPTION
**Pull request by elwoodxblues**
Fixes problems mentioned in code review.

### CODE REVIEW

File: saleor/core/utils/__init__.py [hunk 1]
```diff
@@ -97,3 +97,7 @@ def get_user_shipping_country(request):
         if default_shipping:
             return default_shipping.country
     return request.country
+
+
+def json_dumps_default(obj):
+    return str(obj)
```
Comment (patrys) [inline saleor/core/utils/__init__.py]: Maybe we should check what can actually handle here? I'm afraid we can start to silently produce invalid JSON.

Comment (patrys) [inline saleor/core/utils/__init__.py]: We could also use this to call `price_as_dict` for Price instances.

Comment (artursmet) [inline saleor/core/utils/__init__.py]: I agree, we should use specialized serializer for prices.

**Comment by artursmet** [comment]
@patrys Any other issues in this PR?

**Pull request merged**